torch/_dynamo/guards.py

```diff
@@ -642,7 +685,8 @@ def combine_scopes(left, right):
 
             return {**left, **right}
 
-        def source_ref(source):
+        def source_ref(source, source_list):
+            source_list.append(source)
```

Comment:
I changed my mind, source_list shouldn't be populated here.

The problem is this that interposing on EVERY source reference in produce_guards is too much. Suppose that x has already been inferred to be dynamic, and we have the code:

```
def f(x):
  if x.size(0) > 4:
    return x * 2
  else:
    return x * 3
```

The assert means we will have a guard on "x.size(0) > 4". By your current logic, the code parts for this guard has x.size(0) in its sources. But there is no point in reporting this source to the frame state; x.size(0) is already dynamic, it's not going to change anything to try to make it more dynamic.

So you only want to populate sources for the "constant" guards produced by produce_guards. And we should probably rename sources so that it is more clear that is something specifically about "infer this to be dynamic if this guard fails".

Reply:
Disagree - this system should not know about the other. CodeParts should just collect the correct source for the guard. 

Besides it doesn't matter if this is collected or not - we will just ignore it when the guard for it does not fail during construction of the new object-formerly-known-as-dynamic-plan?